torch/_inductor/memory.py

```diff
@@ -0,0 +1,543 @@
+from __future__ import annotations
+
+import dataclasses
+import heapq
+import logging
+from typing import Callable, Dict, List, Set, Tuple, TYPE_CHECKING, Union
+
+from torch.utils._ordered_set import OrderedSet
+
+from .ir import MultiOutputLayout
+from .utils import get_dtype_size
+from .virtualized import V
+
+
+if TYPE_CHECKING:
+    from .dependencies import Dep
+    from .scheduler import BaseSchedulerNode, SchedulerBuffer
+
+
+torch_log = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class MemoryPlanningInfoForBuffer:
+    size_alloc: int = 0
+    size_free: int = 0
+    outdegree: int = 0
```

Comment:
is `outdegree` only used in LPMF? wonder if we can make it internal to that algorithm function